pkg/activator/certificate/tls_context.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2023 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certificate
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
+
+	"knative.dev/networking/pkg/certificates"
+	pkgnet "knative.dev/pkg/network"
+	"knative.dev/serving/pkg/activator/handler"
+)
+
+// TLSContext returns DialTLSContextFunc.
+func (cr *CertCache) TLSContext() pkgnet.DialTLSContextFunc {
+	return cr.dialTLSContext
+}
+
+// dialTLSContext handles TLS dialer
+func (cr *CertCache) dialTLSContext(ctx context.Context, network, addr string) (net.Conn, error) {
+	return dialTLSContext(ctx, network, addr, cr)
+}
+
+// dialTLSContext handles verify SAN before calling DialTLSWithBackOff.
+func dialTLSContext(ctx context.Context, network, addr string, cr *CertCache) (net.Conn, error) {
+	cr.certificatesMux.Lock()
```

Comment:
defer unlock?

Reply:
At L#47 it unlocks the `certificatesMux`.
We can use `defer` but we just want to lock during the cloning (L#45-46) so unlock at L#47 is enough, I think.